### PR TITLE
[NPU] [Quantization] w4a4 MoE layer support

### PR DESCRIPTION
**Pull request by OrangeRedeng**
## Motivation

- [x] Support w4a4 MoE layer 
- [x] Support w4a4 MoE class in modelslim

For now support only mixed quantization, where w13 (up_gate_proj) in w4a4 and w2 (down_proj) in w8a8. In future will refactor this to add mixed quantization in https://github.com/sgl-project/sglang/pull/17361 PR.

## Modifications

Modify python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py and python/sglang/srt/layers/quantization/modelslim/modelslim_moe.py

## Accuracy Tests

Server:
```ASCEND_RT_VISIBLE_DEVICES=0,1,2,3 python -m sglang.launch_server    --device npu    --attention-backend ascend    --trust-remote-code    --tp-size 4    --model-path /path/to/model/ --mem-fraction-static 0.8 --port 30088```

Client:
```python ./benchmark/gsm8k/bench_sglang.py --num-questions 1319 --port 30088 --data-path ./gsm8k/test.jsonl --parallel 64```

Original:
image

W4A4 quantized:
image

## Benchmarking and Profiling

Client:
```python -m sglang.bench_serving --backend sglang --random-range-ratio 1.0 --dataset-path ./ShareGPT_V3_unfiltered_cleaned_split.json --dataset-name random --num-prompts 64 --max-concurrency 64 --random-input-len 2048 --random-output-len 2048 --host 127.0.0.1 --port 30088```

Original Qwen3-30B-A3B FP16:
image

Qwen3-30B-A3B W4A4-W8A8 mixed:
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py [hunk 1]
```diff
@@ -14,6 +14,94 @@
...
+        **scale_args13,
+        split_item=2,
+        group_list_type=0,
+        group_type=0,
+        group_list=expert_tokens,
+        output_dtype=original_dtype,
+    )[0]
+    # act_fn: swiglu
+    hidden_states = torch.ops.npu.npu_swiglu(hidden_states)
+    hidden_states, pertoken_scale = torch.ops.npu.npu_dynamic_quant(hidden_states)
```
[the remark this answers is not in the document]
Comment (OrangeRedeng) [reply python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py:70]: For now support only mixed quantization, where w13 in w4a4 and w2 in w8a8. In future will refactor this in https://github.com/sgl-project/sglang/pull/17361 PR

File: python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py [hunk 2]
```diff
@@ -148,6 +236,73 @@ def __init__(
...
 
 
+class NPUW4A4Int4DynamicMoEMethod(_NPUFusedMoEMethodBase):
+
+    def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
+        layer.w13_weight.data = layer.w13_weight.data.transpose(1, 2).contiguous()
+        layer.w13_weight.data = npu_format_cast(layer.w13_weight.data)
+        layer.w13_weight.data = self._pack_to_int32(layer.w13_weight.data.to(torch.int32))
+        
+        layer.w2_weight.data = npu_format_cast(layer.w2_weight.data.transpose(1, 2))
```
[the remark this answers is not in the document]
Comment (OrangeRedeng) [reply python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py:324]: For now support only mixed quantization, where w13 in w4a4 and w2 in w8a8. In future will refactor this in https://github.com/sgl-project/sglang/pull/17361 PR

**Review by ping1jing2 — APPROVED** [review]
